x-pack/plugin/esql/qa/security/src/javaRestTest/java/org/elasticsearch/xpack/esql/EsqlSecurityIT.java

```diff
@@ -222,18 +222,60 @@ public void testInsufficientPrivilege() {
     }
 
     public void testLimitedPrivilege() throws Exception {
-        Response resp = runESQLCommand("metadata1_read2", """
-            FROM index-user1,index-user2 METADATA _index
-            | STATS sum=sum(value), index=VALUES(_index)
-            """);
-        assertOK(resp);
-        Map<String, Object> respMap = entityAsMap(resp);
+        ResponseException resp = expectThrows(
+            ResponseException.class,
+            () -> runESQLCommand(
+                "metadata1_read2",
+                "FROM index-user1,index-user2 METADATA _index | STATS sum=sum(value), index=VALUES(_index)"
```

Comment:
Should this work for just
```
                "FROM index-user1,index-user2 METADATA _index | STATS index = VALUES(_index)"
```
because we have metadata privileges on `index-user1`? If so, do we have a test for this as well?

Reply:
@alex-spies it doesn't matter the privileges here, but more how ES discovers the index patterns provided. The change in ComputeService is the essential one where I changed back the indices options to `SearchRequest.DEFAULT_INDICES_OPTIONS` which is slightly different and less permissive than the one we have in `IndexResolver`.

And I _think_ (haven't dug deep into this one) the relevant difference in indices options is 
`.concreteTargetOptions(ConcreteTargetOptions.ALLOW_UNAVAILABLE_TARGETS)` in `IndexResolver` and 
`.concreteTargetOptions(ConcreteTargetOptions.ERROR_WHEN_UNAVAILABLE_TARGETS)` in `SearchRequest`.

What permissions (at Security level) this needs I am not sure, but I tested in the same method of this class the equivalent `_search` query for `from index-user1,index-user2` like this:

```
        XContentBuilder json = JsonXContent.contentBuilder();
        json.startObject();
        json.field("query", QueryBuilders.matchAllQuery());
        json.endObject();
        Request request = new Request("GET", "/index-user1,index-user2/_search");
        request.setJsonEntity(Strings.toString(json));
        request.setOptions(RequestOptions.DEFAULT.toBuilder().addHeader("es-security-runas-user", "metadata1_read2"));
        resp = expectThrows(ResponseException.class, () -> client().performRequest(request));
        assertThat(
            EntityUtils.toString(resp.getResponse().getEntity()),
            containsString(
                "unauthorized for user [test-admin] run as [metadata1_read2] with effective roles [metadata1_read2] on indices [index-user1]"
            )
        );
        assertThat(resp.getResponse().getStatusLine().getStatusCode(), equalTo(HttpStatus.SC_FORBIDDEN));
```

And the test passes which means that ES|QL for this index pattern and test setup (I refer here to users, roles and permissions) behaves exactly like `_search` query DSL does.